lib/vtls/vtls.c

```diff
@@ -1358,6 +1359,13 @@ static CURLcode ssl_cf_connect(struct Curl_cfilter *cf,
     result = Curl_ssl_peer_init(&connssl->peer, cf, tls_id, TRNSPRT_TCP);
     if(result)
       goto out;
+    if(Curl_tls_keylog_enabled()) {
+      infof(data, "SSLKEYLOGFILE set, logging all TLS secrets to '%s'",
+            Curl_tls_keylog_file_name());
+#ifdef LIBRESSL_VERSION_NUMBER
+      infof(data, "Note LibreSSL only supports SSLKEYLOGFILE for TLS <= 1.2");
```

Comment:
LibreSSL supports the keylog API, but they are no-op stubs in
versions released so far.

curl disables `SSLKEYLOGFILE` support for LibreSSL due to this.

Reply:
The usage of `SSL_CTX_set_keylog_callback` is disabled, but for TLS 1.2 it still works, with this code:
https://github.com/curl/curl/blob/dfd781ff627991c7b660fe6534c347b1a97bcc17/lib/vtls/openssl.c#L683-L719

Reply:
Opened https://github.com/curl/everything-curl/pull/572 also